HealthVault/Classes/Caching/Private/MHVThingCacheDatabase.m

```diff
@@ -759,6 +778,231 @@ - (void)updateLastCompletedSyncDate:(NSDate *_Nullable)lastCompletedSyncDate
+                {
+                    completion([NSError MHVCacheError:@"Could not create a new MHVPendingThingOperation."]);
+                }
+                return;
+            }
+            
+        }
+        
+        operation.name = pendingMethod.name;
+        operation.version = pendingMethod.version;
+        operation.originalRequestDate = pendingMethod.originalRequestDate;
+        operation.parameters = pendingMethod.parameters;
+        operation.identifier = pendingMethod.identifier;
+        operation.correlationId = pendingMethod.correlationId.UUIDString;
+        
+        NSError *error = [self saveContext];
+        
+        if (completion)
+        {
+            completion(error);
+        }
+
+    }];
+}
+
+- (void)fetchPendingMethodsForRecordId:(NSString *)recordId
+                            completion:(void (^)(NSArray<MHVPendingMethod *> *_Nullable methods, NSError *_Nullable error))completion
+{
+    if ([NSString isNilOrEmpty:recordId])
+    {
+        if (completion)
+        {
+            completion(nil, [NSError MVHRequiredParameterIsNil]);
+        }
+        return;
+    }
+    
+    if (!self.isDatabaseReady)
+    {
+        if (completion)
+        {
+            completion(nil, [NSError MHVCacheDeleted]);
+        }
+        return;
+    }
+    
+    [self.managedObjectContext performBlock:^
+    {
+        //Create query to filter & order Things
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"record.recordId == %@", [recordId lowercaseString]];
+        
+        NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"MHVPendingThingOperation"];
+        fetchRequest.predicate = predicate;
+        fetchRequest.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"originalRequestDate" ascending:NO]];
+        
+        NSError *error;
+        NSUInteger fetchCount = [self.managedObjectContext countForFetchRequest:fetchRequest error:&error];
+        if (error)
+        {
+            completion(nil, [NSError MHVCacheError:@"Could not calculate the number of pending thing operations."]);
+            return;
+        }
+        
+        NSMutableArray<MHVPendingMethod *> *pendingMethods = [NSMutableArray new];
+        
+        if (fetchCount > 0)
+        {
+            NSArray<MHVPendingThingOperation *> *operations = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+            
+            if (error)
+            {
+                completion(nil, [NSError MHVCacheError:@"Could not fetch things from cache database"]);
+                return;
+            }
+            
+            //Convert cached things back into MHVThings
+            for (int i = 0; i < operations.count; i ++)
+            {
+                MHVPendingThingOperation *operation = operations[i];
+                
+                MHVPendingMethod *pendingMethod = [[MHVPendingMethod alloc] initWithOriginalRequestDate:operation.originalRequestDate
```

Comment:
MHVPendingMethod is a public class - I don't want to create an initializer based on a private class.